src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadHandlerBase.cs

```diff
@@ -37,11 +40,19 @@ protected GetPayloadHandlerBase(int apiVersion, IPayloadPreparationService paylo
             return ResultWrapper<TGetPayloadResult?>.Fail("unknown payload", MergeErrorCodes.UnknownPayload);
         }
 
+        TGetPayloadResult getPayloadResult = GetPayloadResultFromBlock(blockContext);
+
+        if (!getPayloadResult.ValidateFork(_specProvider))
```

Comment:
Why do we need this validation in GetPayload?

Reply:
GetPayload should return payload if it's requested by method of version that assumes particular fork. For example we cannot request Cancun payload using V2 method. And it's hard to validate it earlier, but I guess we can.

Reply:
but we produce those blocks so I am not sure if it is needed at all. We shouldn't have a situation where we produce the wrong block. Don't we?

Reply:
Do we have this check in hive tests or spec?